tests/kernel/smp/src/main.c

```diff
@@ -238,8 +238,14 @@ void test_coop_resched_threads(void)
 	 * since we don't give up current CPU, last thread
 	 * will not get scheduled
 	 */
+#ifdef CONFIG_SCHED_IPI_SUPPORTED
+	/* no need to dealy  if ipi is supported */
+	spawn_threads(K_PRIO_COOP(10), THREADS_NUM, !EQUAL_PRIORITY,
+		      &thread_entry, 0);
+#else
 	spawn_threads(K_PRIO_COOP(10), THREADS_NUM, !EQUAL_PRIORITY,
 		      &thread_entry, THREAD_DELAY);
+#endif
```

Comment:
Can you explain this one better?  I don't understand the bug being fixed.  And regardless: we don't want test code predicated on kernel features unless it's actively testing those features.

If we have test code where timing like this is causing failures, what we almost certainly have is a synchronization bug elsewhere in the test.  Can we fix that instead?

Reply:
* one aspect is :
    For nsim_hs_smp,  i found in test_coop_resched_threads,  threads in other cores didn't run 
    even after k_busy_wait(),  it will make the following zassert_ture failed. 
    
    The key of test_coop_resched_threads is that other cores must run before k_busy_wait. But  this condition cannot be always met (i found both in nsim_hs_smp and hsdk).

So as you said,  we should have a better  synchronization here, but not rely on timing.

* another aspect is that
  After IPI is supported,  there is no need to do tiny delay  which is required by previous busy_wait in idle loop.